tools/test-npm.sh

```diff
@@ -29,10 +29,10 @@ export npm_config_prefix="npm-prefix"
 export npm_config_tmp="npm-tmp"
 
 # install npm devDependencies and run npm's tests
-
-../$NODE cli.js install --ignore-scripts
-../$NODE cli.js run-script test-legacy
-../$NODE cli.js run-script test
+NODEPATH="$(../$NODE -e 'console.log(require("path").resolve(".."))')"
```

Comment:
`node -p` :)

Fwiw I had previously attempted to do something similar in a previous version of PR 1984: https://github.com/nodejs/node/pull/1984#discussion_r33823717

Does the npm patch remove the need for this? I.e. will the new npm patch run without a global node? If so, that would be great for us. :D